### PR TITLE
Visualization Usage Example with HoloViz & GeoDataFrame

### LINKED ISSUE #424: GeoDataFrame Data Structure Usage Example
Create a usage example showcasing the new data structures used for visualization.

### PR DESCRIPTION
**Pull request by philipc2**
Closes #424, #97, #431

## Overview

* Usage example showcasing how to use the `GeoDataFrame` conversion for plotting with HoloViz

## PR Checklist

**General**
- [x] An issue is linked created and linked
- [x] Add appropriate labels
- [x] Filled out Overview and Expected Usage (if applicable) sections

**Examples**
- [x] Any new notebook examples added to `docs/examples/` folder
- [x] Clear the output of all cells before committing
- [x] New notebook files added to `docs/examples.rst` toctree
- [x] New notebook files added to new entry in `docs/gallery.yml` with appropriate thumbnail photo in `docs/_static/thumbnails/`

### CODE REVIEW

**Comment by philipc2** [comment]
Most of the content in this notebook is ready for review. I could use some suggestions for the markdown cells and would appreciate any feedback!

**Review by ifranda — APPROVED** [review]
I like the way you organized this! The workflow seems very clear.

File: docs/examples/005-viz-with-holoviz.ipynb [hunk 1]
```diff
@@ -0,0 +1,431 @@
+{
```
Comment (erogluorhan) [inline docs/examples/005-viz-with-holoviz.ipynb:1]: <p>A few suggestions:</p><ol><li>Re-title: "Conversion to GeoDataFrame for Visualization with Holoviz"</li><li>Authors: We did not have authors section in any other usage example. I'd recommend to be consistent in that way.</li><li>Rephrase the main paragraph: "This notebook showcases how to use UXarray's utility functions for getting its data types (i.e. <code>Grid</code>, <code>UxDataArray</code>) ready to be visualized with HoloViz tools. Specifically, outlined is the conversion to a <code>GeoDataFrame</code> via UXarray's <code>to_geodataframe(), </code>which allows visualization with packages from the HoloViz stack. The notebook showcases basic visualization examples using the <code>hvPlot</code> package after the conversion."</li><li>Please add a note section after the main paragraph and tell that we will have Uxarray's basic plot() function in the near future. Maybe something like: "Conversion to <code>GeoDataFrame</code> works great to use it for visualization afterwards; however, we will release UXarray's own basic plotting function that will support Holoviz tools as backend in the near future, which could make visualization workflows for unstructured grids even more straightforward (i.e. as convenient as calling <code>ux.Grid.plot(...)</code> )"</li></ol>
<br/>

 _Reply via <a href='https://app.reviewnb.com/UXARRAY/uxarray/pull/428/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (erogluorhan) [inline docs/examples/005-viz-with-holoviz.ipynb:1]: <p>Rephrase: "If you wish to plot <em>only</em> the grid's geometrical elements such as nodes, edges (i.e. without any data variables mapped to them), you can use the <code>Grid.to_geodataframe()</code> method to obtain a <code>GeoDataFrame</code> with a singular geometry column having each face represented as a <code>MultiPolygon"</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/UXARRAY/uxarray/pull/428/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (erogluorhan) [inline docs/examples/005-viz-with-holoviz.ipynb:1]: <p>replace "instance" with "interest"?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/UXARRAY/uxarray/pull/428/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (erogluorhan) [inline docs/examples/005-viz-with-holoviz.ipynb:1]: <p>Replace "... Antimeridian (180 degrees east or west), which polygons are incorrectly... " with "Antimeridian (180 degrees east or west) where polygons are incorrectly"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/UXARRAY/uxarray/pull/428/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (erogluorhan) [inline docs/examples/005-viz-with-holoviz.ipynb:1]: <p>Add a paragraph that clarifies things like: "From now on, we have <code>GeoDataFrame</code> to be visualized with Holoviz tools, and there is actually nothing to do with UXarray."</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/UXARRAY/uxarray/pull/428/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Comment (erogluorhan) [inline docs/examples/005-viz-with-holoviz.ipynb:1]: <p>Add a small text or comment to the beginning like "Using <code>matplotlib</code> as the backend"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/UXARRAY/uxarray/pull/428/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

**Comment by philipc2** [comment]
> Thanks very much for putting this together! My overall comments are:
> 
> 1. It'd be great if this notebook communicates that it is not the eventual or only way of visualization with UXarray but provides helper (i.e. conversion) functions that can be used in conjunction with plotting libraries; and in the near future we will provide basic plotting functions that can use different visualization backends, for which we will also have usage examples. So, this example here should leave the doors open for that example.
>    
>    * I've just suggested changes in reviewNB to reflect this phenomenon. Please see the comments in reviewNB for better readibility and correspondence
> 2. Could we please implement a second usage example in this same PR to showcase `to_PolyCollection()` as well? I can help with putting that together if you want.

1. That's a great idea! That was my intention and will reflect that with the markdown.

2. I want to implement the `to_LineCollection` method before making a notebook showcasing the PolyCollection visualization, which we can tackle in a separate PR.

**Comment by aaronzedwick** [comment]
Is the purpose of this to showcase the actually visualization? Or just the needed steps to convert the data?

If the point is to show how to make some kind of visual should there be examples of what it looks like when you plot the data using these methods?

**Comment by philipc2** [comment]
> Is the purpose of this to showcase the actually visualization? Or just the needed steps to convert the data?
It's meant to showcase both

> If the point is to show how to make some kind of visual should there be examples of what it looks like when you plot the data using these methods?

The visuals can be seen if you access the documentation generated from this notebook as seen [here](https://uxarray--428.org.readthedocs.build/en/428/examples/005-viz-with-holoviz.html).

**Comment by aaronzedwick** [comment]
> > Is the purpose of this to showcase the actually visualization? Or just the needed steps to convert the data?
> > It's meant to showcase both
> 
> > If the point is to show how to make some kind of visual should there be examples of what it looks like when you plot the data using these methods?
> 
> The visuals can be seen if you access the documentation generated from this notebook as seen [here](https://uxarray--428.org.readthedocs.build/en/428/examples/005-viz-with-holoviz.html).

That is where I checked, I don't see any visuals. Possible related to this line: gdf_grid = uxds.uxgrid.to_geodataframe() having a recursion error?

**Comment by philipc2** [comment]
> > > Is the purpose of this to showcase the actually visualization? Or just the needed steps to convert the data?
> > > It's meant to showcase both
> > 
> > 
> > > If the point is to show how to make some kind of visual should there be examples of what it looks like when you plot the data using these methods?
> > 
> > 
> > The visuals can be seen if you access the documentation generated from this notebook as seen [here](https://uxarray--428.org.readthedocs.build/en/428/examples/005-viz-with-holoviz.html).
> 
> That is where I checked, I don't see any visuals. Possible related to this line: gdf_grid = uxds.uxgrid.to_geodataframe() having a recursion error?

See #431, working on fixing it right now. Plots should show up once it's resolved

**Comment by philipc2** [comment]
@aaronzedwick 

It should be resolved now.

**Comment by aaronzedwick** [comment]
> @aaronzedwick
> 
> It should be resolved now.

Awesome, I can see it now!

**Comment by philipc2** [comment]
@erogluorhan 

I believe that all of your comments should now be addressed!

I have also added a custom thumbnail for this usage example notebook incase. 

image

**Review by aaronzedwick — APPROVED** [review]
It looks really nice Philip! Very clear and easy to follow. Nice job!

**Pull request merged**